### PR TITLE
document helm value = hubble.dropEventEmitter.enabled

### PR DESCRIPTION
**Pull request by saintdle**
Documenting helm value `hubble.dropEventEmitter.enabled`

### CODE REVIEW

**Review by joestringer — DISMISSED** [review]
Hi @saintdle , thanks for the contribution.

This file is auto-generated from `install/kubernetes/cilium/values.yaml.tmpl`. If you want this to be documented, then the `enabled` field under `dropEventEmitter` in that file needs a comment line above it starting with `#`. After adding the description, you can use the following commands to update all the relevant other files:

```
make -C install/kubernetes
make -C Documentation update-helm-values
```

Additionally, sphinx has a spellchecker that doesn't understand how to interpret configurations like `dropEventEmitter`, so we work around the spellchecker by adding the config option to the dictionary like this:

```
./Documentation/update-spelling_wordlist.sh dropEventEmitter 
```

**Comment by saintdle** [comment]
Thanks for the info, it seems like I'm running into some errors running those commands on macos, I'll try again on a linux machine soon, 

I did check the original helm template and it seems like that line is uncommented, so unsure as to why it's not picked up. 

https://github.com/saintdle/cilium/blob/3a5ec5e5d0c271ff4c4e058df0397b38ebe645a1/install/kubernetes/cilium/values.yaml.tmpl#L1759

**Comment by joestringer** [comment]
> I did check the original helm template and it seems like that line is uncommented, so unsure as to why it's not picked up.

I think it's because there's no comment directly above the configuration item.

**Comment by youngnick** [comment]
@saintdle Is this one still ongoing? Is there anything else you need here?

**Comment by saintdle** [comment]
> @saintdle Is this one still ongoing? Is there anything else you need here?

Just relooked at this and I believe I've done it correctly now as per Joe's comment. 

As I am using MacOS I found the first command provided `make -C install/kubernetes` fails:

```shell
❯ make -C install/kubernetes
: cilium/Chart.yaml
grep -lR -e 'version:' -e 'appVersion:' /Users/dean/Documents/GitHub/cilium/install/kubernetes/cilium/ \
                | xargs -L 1 sed -i -e 's/''\([vV]ersion:\) ''[0-9]\+\.[0-9]\+\.[0-9]\+.*''/\1 '1.17.0-dev'/g'
sed -i 's;icon:.*;icon: https://cdn.jsdelivr.net/gh/cilium/cilium@main/Documentation/images/logo-solo.svg;' "/Users/dean/Documents/GitHub/cilium/install/kubernetes/cilium/Chart.yaml"
sed: 1: "/Users/dean/Documents/G ...": extra characters at the end of d command
make: *** [update-chart] Error 1
```

I found the fix is replacing line 46-50 with the below, not sure if there is a way to have this so using the make file works for both linux and macos easily?:

```shell
update-chart: $(ROOT_DIR)/VERSION
	$(ECHO_GEN)cilium/Chart.yaml
	$(QUIET)grep -lR -e 'version:' -e 'appVersion:' $(CILIUM_CHARTS)/ \
		| xargs -L 1 $(SED) -i '' -e 's/'$(CILIUM_CHART_REGEX)'/\1 '$(VERSION)'/g'
	$(QUIET)$(SED) -i '' 's;icon:.*;icon: $(LOGO_BASE_URL)/cilium@main/$(LOGO_PATH);' $(CHART_FILE)
```

**Comment by joestringer** [comment]
I would guess that some of the steps outlined here could help to convert the sed expression into something cross-platform: https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux .

**Comment by saintdle** [comment]
I could look at that as a separate issue :)

**Comment by saintdle** [comment]
tried to squash the commits and add the sign off by, but keep hitting an issue. Unsure how to resolve.

**Review by aanm — APPROVED** [review]

**Review by gandro — APPROVED** [review]
Thank you!

**Review by learnitall — APPROVED** [review]
Thanks!

**Comment by learnitall** [comment]
It looks like the workflow is failing because the sign-off has a typo:

```
<22192242+saintdle@users.noreply.github.com
```

It's missing the `>` character at the end. @saintdle did you add the sign-off manually?

**Comment by saintdle** [comment]
> It looks like the workflow is failing because the sign-off has a typo:
> 
> 
> 
> ```
> 
> <22192242+saintdle@users.noreply.github.com
> 
> ```
> 
> 
> 
> It's missing the `>` character at the end. @saintdle did you add the sign-off manually?

Yep

**Review by tommyp1ckles — APPROVED** [review]

**Review by joestringer — COMMENTED** [review]
Changes LGTM but the sign-off needs to be fixed. Will mark as draft until that is addressed. Once you've fixed that up, please click "Ready for Review" and we can retrigger tests and merge this.

**Comment by saintdle** [comment]
> Changes LGTM but the sign-off needs to be fixed. Will mark as draft until that is addressed. Once you've fixed that up, please click "Ready for Review" and we can retrigger tests and merge this.

Hi, clicked ready for review as finally fixed the commit message, it wanted the sign off piece on a separate line, which is automatic when you use `git commit -s`. 

But now it's pulled through a missing other changes to files from main it seems? But I didn't hit rebase etc. Can you help me work through fixing that bit?

**Comment by joestringer** [comment]
The easiest way is probably to walk back to an earlier copy and then rebase that and address the feedback afresh. If all of the latest changes have been merged into the patch, then it is likely to be tedious work to untangle that. 

Scanning up through the PR I see this: 
image

I would suggest going back to https://github.com/cilium/cilium/commit/73333666fa4e802f6eb51deedcceab9ef6782de9 and then rebase onto the latest `main` branch with `git rebase`, then amend the commit with the recent changes that you made.

**Comment by saintdle** [comment]
> The easiest way is probably to walk back to an earlier copy and then rebase that and address the feedback afresh. If all of the latest changes have been merged into the patch, then it is likely to be tedious work to untangle that.
> 
> Scanning up through the PR I see this: image
> 
> I would suggest going back to [7333366](https://github.com/cilium/cilium/commit/73333666fa4e802f6eb51deedcceab9ef6782de9) and then rebase onto the latest `main` branch with `git rebase`, then amend the commit with the recent changes that you made.

Thanks, will do that tomorrow, and moved back to draft, so that I can fix this up :)

**Comment by saintdle** [comment]
> The easiest way is probably to walk back to an earlier copy and then rebase that and address the feedback afresh. If all of the latest changes have been merged into the patch, then it is likely to be tedious work to untangle that.
> 
> Scanning up through the PR I see this: image
> 
> I would suggest going back to [7333366](https://github.com/cilium/cilium/commit/73333666fa4e802f6eb51deedcceab9ef6782de9) and then rebase onto the latest `main` branch with `git rebase`, then amend the commit with the recent changes that you made.

It seems like I've fixed some of it, but not all of it, unfortunately, and I still have conflicts with two files. I'd really like to understand what I'm doing wrong here so I can fix it.

File: .nvim/nvim-dap.lua [hunk 1]
```diff
@@ -1,15 +1,5 @@
...
-dap.adapters.cilium_delve = {
-  type = 'server',
-  port = '${port}',
-  executable = {
-    command = 'dlv',
-    args = {'dap', '-l', '127.0.0.1:${port}'},
-    -- add this if on windows, otherwise server won't open successfully
-    -- detached = false
-  }
-}
```
Comment (joestringer) [inline .nvim/nvim-dap.lua:12]: It looks like this patch has a bunch of changes merged in that are unrelated to the change you're trying to make (probably you amended the commits to combine a merge commit plus your intended changes at some point). 

If you check out https://github.com/cilium/cilium/commit/73333666fa4e802f6eb51deedcceab9ef6782de9 and fix that one up for any feedback plus do a git rebase, these changes should go away. You can then force push your new copy of the branch over the top of the version in your repository that this PR is being opened from.

**Pull request closed without merging**